python/lib/mesh.cc

```diff
@@ -2,126 +2,315 @@
 #include "framework/mesh/mesh_generator/extruder_mesh_generator.h"
 #include "framework/mesh/mesh_generator/orthogonal_mesh_generator.h"
 #include "framework/mesh/mesh_generator/from_file_mesh_generator.h"
 #include "framework/mesh/mesh_generator/split_file_mesh_generator.h"
 #include "framework/mesh/mesh_generator/distributed_mesh_generator.h"
-#include "framework/graphs/graph_partitioner.h"
-#include "framework/graphs/kba_graph_partitioner.h"
-#include "framework/graphs/linear_graph_partitioner.h"
-#include "framework/graphs/petsc_graph_partitioner.h"
-#include <memory>
+#include "framework/mesh/surface_mesh/surface_mesh.h"
 
 namespace opensn
 {
 
-// clang-format off
-
-void wrap_mesh(py::module &mesh)
+// Wrap mesh continuum
+void WrapMesh(py::module& mesh)
 {
+  // Mesh continuum
   auto mesh_continuum = py::class_<MeshContinuum, std::shared_ptr<MeshContinuum>>(
-    mesh, "MeshContinuum",
+    mesh,
+    "MeshContinuum",
     R"(
-      Mesh continuum.
+    Mesh continuum.
 
-      Wrapper of :cpp:class:`opensn::MeshContinuum`.
-    )");
+    Wrapper of :cpp:class:`opensn::MeshContinuum`.
+    )"
+  );
+  mesh_continuum.def_property(
+    "dimension",
+    &MeshContinuum::GetDimension,
+    &MeshContinuum::SetDimension,
+    "Number of dimensions of the mesh."
+  );
+  mesh_continuum.def(
+    "SetUniformMaterialID",
+    &MeshContinuum::SetUniformMaterialID,
+    "Set material ID's for all cells to the specified material ID.",
+    py::arg("mat_id")
+  );
+  mesh_continuum.def(
+    "SetMaterialIDFromLogicalVolume",
+    &MeshContinuum::SetMaterialIDFromLogical,
+    R"(
+    Set material ID's using a logical volume.
 
-  mesh_continuum.def_property("dimension", &MeshContinuum::GetDimension, &MeshContinuum::SetDimension,
-                              "Number of dimensions of the mesh.");
-  mesh_continuum.def("SetUniformMaterialID", &MeshContinuum::SetUniformMaterialID,
-                     "Set material ID's for all cells to the specified material ID.");
-  mesh_continuum.def("SetMaterialIDFromLogical", &MeshContinuum::SetMaterialIDFromLogical,
-                     "Set material ID's using a logical volume.");
-  mesh_continuum.def("SetBoundaryIDFromLogical", &MeshContinuum::SetBoundaryIDFromLogical,
-                     "Set boundary ID's using a logical volume.");
-  mesh_continuum.def("SetupOrthogonalBoundaries", &MeshContinuum::SetupOrthogonalBoundaries, "...");
+    Parameters
+    ----------
+    log_vol: pyopensn.logvol.LogicalVolume
+        ???
+    mat_id: int
+        ???
+    sense: bool
+        ???
+    )",
+    py::arg("log_vol"),
+    py::arg("mat_id"),
+    py::arg("sense")
+  );
+  mesh_continuum.def(
+    "SetBoundaryIDFromLogicalVolume",
+    &MeshContinuum::SetBoundaryIDFromLogical,
+    R"(
+    Set boundary ID's using a logical volume.
 
+    Parameters
+    ----------
+
+    log_vol: pyopensn.logvol.LogicalVolume
+        ???
+    boundary_name: str
+        ???
+    sense: bool
+        ???
```

Comment:
similar to my previous comments on the setmaterialIDfrom logical volume, except for boundary IDs. now it will be whether the face centers are inside or not of the logival volume to assign a boundaryID. also change `sense` to `inside`